### PR TITLE
Internalizes the IR used by PlanTyper

### PR DESCRIPTION
**Pull request by RCHowell**
## Description
This PR removes all "unresolved" aspects of the plan from the public API. Customers need not filter on such nodes, and we are able to decouple the PlanTyper internals from the public API.

This works by a transform (PlanTransform) at the end of planning which dumps the public API. As of now, the internal unresolved plan and public Plan API are similar so it's a repetitive class. There has been discussion on means of better structures to use for typing — which we are free to experiment with as the internals are no longer tied to the API. 

## Other Information
- Updated Unreleased Section in CHANGELOG: **[YES/NO]**
No

- Any backward-incompatible changes? **[YES/NO]**
No, partiql-planner branch is unreleased

- Any new external dependencies? **[YES/NO]**
No

- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **[YES/NO]**
Yes

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by johnedquinn — COMMENTED** [review]
Very small comments, but beyond that, looks good to me. Thanks for this clean up!

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+        }
+
+        private fun Identifier.normalize(): String = when (this) {
+            is Identifier.Qualified -> (listOf(root.normalize()) + steps.map { it.normalize() }).joinToString(".")
+            is Identifier.Symbol -> when (caseSensitivity) {
+                Identifier.CaseSensitivity.SENSITIVE -> symbol
+                Identifier.CaseSensitivity.INSENSITIVE -> symbol.lowercase()
+            }
+        }
+    }
```
[the remark this answers is not in the document]
Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt:418]: It's just for a debug message, but I can include the quotes

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+                    sourceLocation = UNKNOWN_PROBLEM_LOCATION,
+                    details = PlanningProblemDetails.UndefinedVariable(
+                        variableName = variable.identifier.normalize(),
+                        caseSensitive = false
+                    )
+                )
+            )
+        }
+
+        private fun handleUnknownFunction(onProblem: ProblemCallback, fn: Fn.Unresolved) {
```
[the remark this answers is not in the document]
Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt:399]: They aren't and should be removed. I will clean this up

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+    @OptIn(PartiQLValueExperimental::class)
+    override fun visitRexOpLit(node: Rex.Op.Lit, ctx: ProblemCallback) = org.partiql.plan.rexOpLit(node.value)
+
+    override fun visitRexOpVar(node: Rex.Op.Var, ctx: ProblemCallback) = super.visitRexOpVar(node, ctx) as org.partiql.plan.Rex.Op
+
+    override fun visitRexOpVarResolved(node: Rex.Op.Var.Resolved, ctx: ProblemCallback) =
+        org.partiql.plan.Rex.Op.Var(node.ref)
+
+    override fun visitRexOpVarUnresolved(node: Rex.Op.Var.Unresolved, ctx: ProblemCallback) =
+        org.partiql.plan.Rex.Op.Err("Unresolved function $node")
```
[the remark this answers is not in the document]
Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/PlanTransform.kt:108]: ack

**Review by johnedquinn — APPROVED** [review]

**Pull request merged**